api/app/db/crud/api_access_audits.py

```diff
@@ -14,9 +14,13 @@ def create_api_access_audit_log(
         success: bool,
         path: str) -> None:
     """ Create an audit log. """
-    with app.db.database.get_write_session_scope() as session:
-        now = get_utc_now()
-        audit_log = APIAccessAudit(create_user=username, path=path, success=success,
-                                   create_timestamp=now)
-        session.add(audit_log)
-        session.commit()
+    try:
+        with app.db.database.get_write_session_scope() as session:
+            now = get_utc_now()
+            audit_log = APIAccessAudit(create_user=username, path=path, success=success,
+                                       create_timestamp=now)
+            session.add(audit_log)
+            session.commit()
+    except Exception as exception:
+        logger.error(exception, exc_info=True)
+        raise
```

Comment:
Codecov patch check wants a test for this case, kind of annoying but easy to add I think

Reply:
Oh dear me - except that I forgot how to write tests, and that took me forever. Hopefully we're good now.